crud/update.lua

```diff
@@ -21,14 +22,63 @@ local function update_on_storage(space_name, key, operations)
         return nil, UpdateError:new("Space %q doesn't exist", space_name)
     end
 
-    local tuple = space:update(key, operations)
-    return tuple
+    -- add_space_schema_hash is false because
+    -- reloading space format on router can't avoid update error on storage
+    return schema.wrap_box_space_func_result(false, space, 'update', key, operations)
 end
 
 function update.init()
    _G._crud.update_on_storage = update_on_storage
 end
 
+-- returns result, err, need_reload
+-- need_reload indicates if reloading schema could help
+-- see crud.common.schema.wrap_func_reload()
+local function call_update(space_name, key, user_operations, opts)
+    dev_checks('string', '?', 'table', {
+        timeout = '?number',
+        bucket_id = '?number|cdata',
+    })
+
+    opts = opts or {}
+
+    local space = utils.get_space(space_name, vshard.router.routeall())
+    if space == nil then
+        return nil, UpdateError:new("Space %q doesn't exist", space_name), true
+    end
+
+    local space_format = space:format()
+
+    if box.tuple.is(key) then
+        key = key:totable()
+    end
+
+    local operations, err = utils.convert_operations(user_operations, space_format)
+    if err ~= nil then
+        return nil, UpdateError:new("Wrong operations are specified: %s", err), true
+    end
+
+    local bucket_id = sharding.key_get_bucket_id(key, opts.bucket_id)
+    local storage_result, err = call.rw_single(
+        bucket_id, UPDATE_FUNC_NAME, {space_name, key, operations},
+        {timeout=opts.timeout})
```

Comment:
Since you anyway change this piece of code, you have a great chance to add spaces around `=` as for other places.

Reply:
Who you are? Oleg always suggests doing such things in a separated patch.